ui-src/index.js

```diff
@@ -74,25 +74,15 @@ function req(url) {
  * @param {string} format Format string
  * @return {string}
  */
-function formatTime(ts90k, format = 'YYYY-MM-DDTHH:mm:ss.SSSSSZ') {
+function formatTime(ts90k, format = 'YYYY-MM-DDTHH:mm:ss:FFFFFFZ') {
   const ms = ts90k / 90.0;
   let msRound = 0.5;
-  // Check if format string requires subsecond precision beyond ms
-  const matches = format.match(/(S+)/);
-  if (matches && (matches[1].length > 3)) {
-    // Determine portion not handled by moment, which uses ms
-    const fracms = ts90k % 90000; // Fractional ms, 90k units
-    console.log('fracms: ' + fracms);
-    if (fracms >= 0.001) { // Nothing extra needed if fraction is < 1 µs
-      const subFmt = matches[1];
-      const extraLen = Math.min(subFmt.length - 3, 3); // µs precision
-      const fracµs = fracms / 0.09 + // µs units
-        (0.0 * Math.pow(10, (3 - extraLen))); // and rounding
-      const newSubFmt = subFmt.substr(0, 3) +
-        String(1000000 + fracµs).substr(4, extraLen);
-      format = format.replace(subFmt, newSubFmt);
-      msRound = 0;
-    }
+  const fractionFormat = 'FFFFF';
+  const fmtRegex = new RegExp('(' + fractionFormat + ')', 'g');
```

Comment:
I haven't benchmarked this, but I think it compiles the regular expression every time formatTime is run. regex compilation is quite slow (typically orders of magnitude slower than matching).

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions suggests using

const fmtRegex = /(FFFFF)/;

(the 'g' modifier doesn't do anything; you're only looking once.)

Reply:
btw, an alternative would be to not use regexes at all:

```js
function formatTime(ts90k, format = 'YYYY-MM-DDTHH:mm:ss:FFFFFZ') {
  const ms = ts90k / 90.0;
  const fracFmt = 'FFFFF';
  let fracLoc = format.indexOf(fracFmt);
  if (fracLoc != -1) {
    const frac = ts90k % 90000;
    format = format.substr(0, fracLoc) + String(100000 + frac).substr(1) +
             format.substr(fracLoc + fracFmt.length);
  }
  return moment.tz(ms, zone).format(format);
}
```

Reply:
Yep, you are right.